subsys/bluetooth/gatt_dm.c

```diff
@@ -377,7 +389,11 @@ static uint8_t discovery_process_characteristic(
 	struct bt_gatt_chrc *cur_gatt_chrc;
 
 	if (!attr) {
-		discovery_complete(dm);
+		if (!dm->chrc_to_store) {
```

Comment:
This case covers only partially disconnection scenario. If disconnection is after `discovery_process_characteristic` and before `discovery_process_attribute`, we will not report an error. Crash will not happen in this case, but it looks like an inconsistent behavior.

Reply:
Right. It seems that calling `discovery_complete_error` is unnecessary if we take care of the validity of the uuid pointers.